stake-pool/cli/src/main.rs

```diff
@@ -143,299 +230,343 @@ fn command_create_account(config: &Config, token: Pubkey) -> CommandResult {
+        println!(
+            "Not enough token balance to withdraw {} SOL.\nMaximum withdraw amount is {} SOL.",
+            lamports_to_sol(amount),
+            lamports_to_sol(max_withdraw_amount)
+        );
+        return Ok(None);
+    }
 
-fn command_accounts(config: &Config, token: Option<Pubkey>) -> CommandResult {
-    let accounts = config
-        .rpc_client
-        .get_token_accounts_by_owner_with_commitment(
-            &config.owner.pubkey(),
-            match token {
-                Some(token) => TokenAccountsFilter::Mint(token),
-                None => TokenAccountsFilter::ProgramId(spl_token::id()),
-            },
-            config.commitment_config,
-        )?
-        .value;
+    let mut accounts = get_authority_accounts(config, &pool_withdraw_authority);
     if accounts.is_empty() {
-        println!("None");
+        println!("No accounts found.");
+        return Ok(None);
     }
+    // Sort from lowest to highest balance
+    accounts.sort_by(|a, b| a.1.lamports.cmp(&b.1.lamports));
 
-    println!("Account                                      Token                                        Balance");
-    println!("-------------------------------------------------------------------------------------------------");
-    for keyed_account in accounts {
-        let address = keyed_account.pubkey;
-
-        if let UiAccountData::Json(parsed_account) = keyed_account.account.data {
-            if parsed_account.program != "spl-token" {
-                println!(
-                    "{:<44} Unsupported account program: {}",
-                    address, parsed_account.program
-                );
-            } else {
-                match serde_json::from_value(parsed_account.parsed) {
-                    Ok(TokenAccountType::Account(ui_token_account)) => println!(
-                        "{:<44} {:<44} {}",
-                        address, ui_token_account.mint, ui_token_account.token_amount.ui_amount
-                    ),
-                    Ok(_) => println!("{:<44} Unsupported token account", address),
-                    Err(err) => println!("{:<44} Account parse failure: {}", address, err),
-                }
-            }
+    for (stake_pubkey, account) in accounts {
+        if account.lamports < amount {
+            continue;
+        }
+        // Account found to claim or withdraw
+        println!("Withdrawing from account {}", stake_pubkey);
+
+        let mut instructions: Vec<Instruction> = vec![];
+        let mut signers = vec![config.fee_payer.as_ref(), config.owner.as_ref()];
+
+        let stake_receiver_account = Keypair::new();
+
+        let mut total_rent_free_balances: u64 = 0;
+
+        // Calculate amount of tokens to burn
+        let tokens_to_burn = (amount as u128)
+            .checked_mul(pool_data.pool_total as u128)
+            .unwrap()
+            .checked_div(pool_data.stake_total as u128)
+            .unwrap() as u64;
```

Comment:
For ease of development, this can get factored into a different function, since it's also being used on line 431

Reply:
Any answer here?

Reply:
Actually there are two opposite calculations (amount to token on 468 and token to amount on 431), will make them separate functions.

Reply:
Ah right, my mistake!  I get confused between stake tokens and pool tokens